### PR TITLE
Profile settings

### PR DESCRIPTION
**Pull request by likeri29**
## Describe your changes
- I added the possibility for a user to delete his/her account
- I added the possibility for a user to toggle the notifications
- I added the possibility for a user to delete his/her account

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-307)
  
### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/users/users.service.ts [hunk 1]
```diff
@@ -437,22 +438,27 @@ export class UsersService {
     }
   }
 
-  async softDeleteUser(id: string): Promise<void> {
+  async softDeleteUser(idToDelete: string, userId: string): Promise<void> {
     try {
-      const user = await this.getUserById(id);
+      const userToDelete = await this.getUserById(idToDelete);
+      const user = await this.getUserById(userId);
```
Comment (lyudad) [inline src/modules/users/users.service.ts]: but we are sure that user exist, we check it in the validation guard, no?

File: src/modules/users/users.service.ts [hunk 2]
```diff
@@ -503,12 +510,61 @@ export class UsersService {
     try {
       const hashedPassword = await this.hashPassword(password);
 
+      const user = await this.getUserById(id);
```
Comment (lyudad) [inline src/modules/users/users.service.ts]: again from my understanding we validate it in step above, no?

File: src/modules/users/users.service.ts (hunk 1 above)
Comment (likeri29) [reply src/modules/users/users.service.ts]: But below I don't check if the user exists, I need this to check if the userToDelete has the same ID as user or has the admin role to delete user. In my opinion I should have these lines of code

File: src/modules/users/users.service.ts (hunk 2 above)
Comment (likeri29) [reply src/modules/users/users.service.ts]: I agree, I deleted the if statement, but I need the user to get his email and name to send an email notification about password change

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -290,4 +299,45 @@ export class AuthService {
       return newUser;
     }
   }
+
+  async changeEmail(id: string, email: string): Promise<void> {
+    try {
+      const user = await this.usersService.getUserById(id);
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: you should have user object from guard where you validate token, you don't need to make query again

File: src/modules/users/users.service.ts (hunk 1 above)
Comment (lyudad) [reply src/modules/users/users.service.ts]: why line 444 needed? you should already have it from the guard

**Review by lyudad — APPROVED** [review]

**Pull request merged**